network/server/tchannelthrift/node/service.go

```diff
@@ -478,6 +485,139 @@ func (s *service) FetchBlocksMetadataRaw(tctx thrift.Context, req *rpc.FetchBloc
 	return result, nil
 }
 
+func (s *service) FetchBlocksMetadataRawV2(tctx thrift.Context, req *rpc.FetchBlocksMetadataRawV2Request) (*rpc.FetchBlocksMetadataRawV2Result_, error) {
+	if s.db.IsOverloaded() {
+		s.metrics.overloadRejected.Inc(1)
+		return nil, tterrors.NewInternalError(errServerIsOverloaded)
+	}
+
+	var err error
+	callStart := s.nowFn()
+	defer func() {
```

Comment:
Is this path "cold" enough to use a defer?

Reply:
Single defer is not so bad, it's been sped up in 1.8 https://golang.org/doc/go1.8#defer - it's a batch call so it's probably ok.  

Although you shouldn't always `ReportSuccess(...)` you can use "ReportSuccessOrError(...)` if you want.